File: Medium/2026-08-26-2904-Shortest-and-Lexicographically-Smallest-Beautiful-String/solution.java
class Solution {
    public String smallestBeautifulString(String s, int k) {
        int n = s.length();
        String result = "";
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j <= n; j++) {
                String substring = s.substring(i, j);
                if (countOnes(substring) == k) {
                    if (result.isEmpty() || substring.length() < result.length() || 
                        (substring.length() == result.length() && substring.compareTo(result) < 0)) {
                        result = substring;
                    }
                }
            }
        }
        
        return result;
    }
    
    private int countOnes(String s) {
        int count = 0;
        for (char c : s.toCharArray()) {
            if (c == '1') {
                count++;
            }
        }
        return count;
    }
}

File: Medium/2026-08-26-2904-Shortest-and-Lexicographically-Smallest-Beautiful-String/solution.py
class Solution:
    def smallestBeautifulString(self, s: str, k: int) -> str:
        n = len(s)
        min_length = float('inf')
        result = ""
        
        for i in range(n):
            count_ones = 0
            for j in range(i, n):
                if s[j] == '1':
                    count_ones += 1
                if count_ones > k:
                    break
                if count_ones == k:
                    substring = s[i:j + 1]
                    if len(substring) < min_length or (len(substring) == min_length and substring < result):
                        min_length = len(substring)
                        result = substring
        
        return result